fuel/transformers/__init__.py

```diff
@@ -257,6 +276,44 @@ def get_data(self, request=None):
                 raise
         return tuple(numpy.asarray(source_data) for source_data in data)
 
+    def get_example(self, request=None):
+        raise AttributeError(str(type(self)) + 
+        "is a batch method only")
+
+    def get_batch(self,request=None):
+        return self.get_data(request)
+
+
+class Unpack(Transformer):
+    """Unpacks batches to compose a stream of examples.
+    This class is the inverse of the Batch class: it turns a minibatch into
+    a stream of examples.
+    Parameters
+    ----------
+    data_stream : :class:`AbstractDataStream` instance
+        The data stream to unpack
+    """
+    def __init__(self, data_stream, batch=False):
+        super(Unpack, self).__init__(data_stream, batch)
+        self.data = None
+
+    def get_data(self, request=None):
+        if not self.data:
+            data = next(self.child_epoch_iterator)
+            self.data = izip(*data)
+        try:
+            return next(self.data)
+        except StopIteration:
+            self.data = None
+            return self.get_data()
+
+    def get_example(self, request=None):
+        return self.get_data(request)
+
+    def get_batch(self,request=None):
```

Comment:
Instead of doing this for each transformer, implement a `get_batch` and `get_example` method that raises a `NotImplementedError` in the `Transformer` base class.